tiny_dnn/core/framework/tensor.h

```diff
@@ -128,16 +128,22 @@ class TensorStorage {
  */
 template<typename U = float_t, size_t kDimensions = 4, bool kConst = false>
```

Comment:
Adding the constness as a template parameter is a bit confusing for me.

Can't the uses of kConst be changed into is_const_v<U>? (Or U be forwarded to the underlying TensorStorage, to make it type-safe?)